src/compiler/program.ts

```diff
@@ -70,7 +70,7 @@ namespace ts {
     }
 
     function loadNodeModuleFromFile(candidate: string, failedLookupLocation: string[], host: ModuleResolutionHost): string {
-        return forEach(moduleFileExtensions, tryLoad);
+        return forEach(supportedJsExtensions, tryLoad);

```

Comment:
this is a good question. I do remember that we want to lift restriction for allow only `.d.ts` files and add `.ts` / `.tsx` to the mix but I'm not sure about `.js`